PyDynamic/uncertainty/propagate_DFT.py

```diff
@@ -1,776 +1,908 @@
-		UF: np.ndarray
-			covariance matrix associated with F
-
-	"""
-
-	assert(len(A.shape)==1)
-	assert(A.shape==P.shape)
-	assert(UAP.shape==(2*len(A),2*len(A)) or UAP.shape==(2*len(A),))
-	F = np.r_[A*np.cos(P),A*np.sin(P)]		# calculation of best estimate
-
-	# calculation of sensitivities
-	CRA = np.cos(P)
-	CRP = -A*np.sin(P)
-	CIA = np.sin(P)
-	CIP = A*np.cos(P)
-
-	# assignment of uncertainty blocks in UAP
-	N = len(A)
-	if UAP.shape==(2*N,): # zero correlation; just standard deviations
-		Ua = UAP[:N]
-		Up = UAP[N:]
-		U11 = CRA*Ua*CRA + CRP*Up*CRP
-		U12 = CRA*Ua*CIA + CRP*Up*CIP
-		U22 = CIA*Ua*CIA + CIP*Up*CIP
-		UF = sparse.diags([np.r_[U11,U22],U12,U12],[0,N,-N])
-		if not keep_sparse:
-			UF = UF.toarray()
-	else:
-		if isinstance(UAP,sparse.dia_matrix):
-			nrows = 2*N
-			offset= UAP.offsets
-			diags = UAP.data
-			Uaa = diags[0][:N]
-			Uap = diags[1][offset[1]:nrows+offset[1]]
-			Upp = diags[0][N:]
-
-			U11 = Uaa*CRA**2 + CRP*Uap*CRA + CRA*Uap*CRP + Upp*CRP**2
-			U12 = CRA*Uaa*CIA + CRP*Uap*CIA + CRA*Uap*CIA + CRP*Upp*CIP
-			U22 = Uaa*CIA**2 + CIP*Uap*CIA + CIA*Uap*CIP + Upp*CIP**2
-
-			UF = sparse.diags([np.r_[U11,U22],U12,U12],[0,N,-N])		# default is sparse
-			if not keep_sparse:
-				UF = UF.toarray()		# fall back to non-sparse
-		else:
-			Uaa = UAP[:N,:N]
-			Uap = UAP[:N,N:]
-			Upp = UAP[N:,N:]
-
-			U11 = prod(CRA,prod(Uaa,CRA)) + prod(CRP,prod(Uap.T,CRA)) + prod(CRA,prod(Uap,CRP)) + prod(CRP,prod(Upp,CRP))
-			U12 = prod(CRA,prod(Uaa,CIA)) + prod(CRP,prod(Uap.T,CIA)) + prod(CRA,prod(Uap,CIA)) + prod(CRP,prod(Upp,CIP))
-			U22 = prod(CIA,prod(Uaa,CIA)) + prod(CIP,prod(Uap.T,CIA)) + prod(CIA,prod(Uap,CIP)) + prod(CIP,prod(Upp,CIP))
-
-			UF = np.vstack((np.hstack((U11,U12)),np.hstack((U12.T,U22))))	# stack together the full covariance matrix
-
-	return F, UF
-
-def Time2AmpPhase(x,Ux):
-	"""Transformation from time domain to amplitude and phase
-
-	Parameters
-	----------
-		 x: np.ndarray of shape (N,)
-		  	time domain signal
-		 Ux: np.ndarray of shape (N,N)
-		 	squared uncertainty associated with x
-
-	Returns
-	-------
-		A: np.ndarray
-			amplitude values
-		P: np.ndarray
-			phase values
-		UAP: np.ndarray
-			covariance matrix associated with [A,P]
-	"""
-	F,UF = GUM_DFT(x,Ux)			# propagate to DFT domain
-	A,P,UAP = DFT2AmpPhase(F,UF)	# propagate to amplitude and phase
-	return A,P,UAP
+    """Return the Discrete Fourier Transform sample frequencies
+
+    Parameters
+    ----------
+        N: int
+            window length
+        dt: float
+            sample spacing (inverse of sampling rate)
+
+    Returns
+    -------
+        f: ndarray
+            Array of length ``n//2 + 1`` containing the sample frequencies
+
+    See also
+    --------
+        `mod`::numpy.fft.rfftfreq
+
+    """
+
+    return np.fft.rfftfreq(N, dt)
+
+
+def DFT2AmpPhase(F, UF, keep_sparse=False, tol=1.0, return_type="separate"):
+    """Transformation from real and imaginary parts to magnitude and phase
```

Comment:
In many of the other functions, >>"""<< was followed by a newline. Maybe here to?